src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -198,6 +198,15 @@ private TypeSpec CreateTypeSpec(TypeParseInfo typeParseInfo)
                 else if (IsCollection(type))
                 {
                     spec = CreateCollectionSpec(typeParseInfo);
+
+                    // fallback to treating as an object if we couldn't treat as a collection and
+                    // it does not implement ICollection<> nor IDictionary<,>
+                    if (spec is UnsupportedTypeSpec && type is INamedTypeSymbol namedType &&
+                        GetInterface(namedType, _typeSymbols.GenericICollection_Unbound) is null &&
```

Comment:
Do these interfaces also get resolved inside the `CreateCollectionSpec` method? If so, would it be worth avoiding additional a bit of duplicated work by returning an `out bool doesNotImplementKnownMutableCollectionInterfaces` parameter? Or perhaps embed that information in the `UnsupportedTypeSpec` for a cleaner look?